modules/python/Makefile.am

```diff
@@ -1,75 +1,97 @@
-	modules/python/compat/compat-python.h     \
-	modules/python/compat/compat-python.c     \
-	modules/python/python-module.h		  \
-	modules/python/python-config.h		  \
-	modules/python/python-config.c		  \
-	modules/python/python-helpers.h		  \
-	modules/python/python-helpers.c		  \
-	modules/python/python-main.h		  \
-	modules/python/python-main.c		  \
-	modules/python/python-plugin.c		  \
-	modules/python/python-grammar.y		  \
-	modules/python/python-value-pairs.c	  \
-	modules/python/python-value-pairs.h	  \
-	modules/python/python-dest.c		  \
-	modules/python/python-dest.h		  \
-	modules/python/python-tf.c		  \
-	modules/python/python-tf.h		  \
-	modules/python/python-parser.c		  \
-	modules/python/python-parser.h	          \
-	modules/python/python-logmsg.h		  \
-	modules/python/python-logmsg.c	          \
-	modules/python/python-logtemplate.h	  \
-	modules/python/python-logtemplate.c       \
-	modules/python/python-logtemplate-options.h\
-	modules/python/python-logtemplate-options.c\
-	modules/python/python-integerpointer.c    \
-	modules/python/python-integerpointer.h    \
-	modules/python/python-logger.c    \
-	modules/python/python-logger.h    \
-	modules/python/python-global-code-loader.h\
-	modules/python/python-global-code-loader.c\
-	modules/python/python-debugger.h	  \
-	modules/python/python-debugger.c	  \
-	modules/python/python-logparser.h	  \
-	modules/python/python-logparser.c	  \
-	modules/python/python-source.h	  \
-	modules/python/python-source.c	  \
-	modules/python/python-fetcher.h	  \
-	modules/python/python-fetcher.c
+python_common_cppflags = \
+	$(AM_CPPFLAGS) \
+	-I$(top_srcdir)/modules/python \
+	-I$(top_builddir)/modules/python
 
-modules_python_libmod_python_la_LDFLAGS		 = \
-	$(MODULE_LDFLAGS) \
-	$(PYTHON_LDADD)
+if !ENABLE_PYTHONv2_AND_v3
 
-modules_python_libmod_python_la_DEPENDENCIES=	\
-	$(MODULE_DEPS_LIBS)
+module_LTLIBRARIES += modules/python/libmod-python.la
+modules_python_libmod_python_la_CFLAGS = $(python_common_cflags) $(PYTHON_CFLAGS)
+modules_python_libmod_python_la_CPPFLAGS = $(python_common_cppflags)
+modules_python_libmod_python_la_LIBADD = $(python_common_libadd) $(PYTHON_LIBS)
+modules_python_libmod_python_la_SOURCES = $(python_sources)
+modules_python_libmod_python_la_LDFLAGS = $(MODULE_LDFLAGS) $(PYTHON_LDADD)
+modules_python_libmod_python_la_DEPENDENCIES = $(MODULE_DEPS_LIBS)
 
 modules/python mod-python: modules/python/libmod-python.la
 
+else
+
+module_LTLIBRARIES += modules/python/libmod-python.la
+modules_python_libmod_python_la_CFLAGS = $(python_common_cflags) $(PYTHON2_CFLAGS)
+modules_python_libmod_python_la_CPPFLAGS = -DSYSLOG_NG_ENABLE_PYTHONv2=1 $(python_common_cppflags)
+modules_python_libmod_python_la_LIBADD = $(python_common_libadd) $(PYTHON2_LIBS)
+modules_python_libmod_python_la_SOURCES = $(python_sources)
+modules_python_libmod_python_la_LDFLAGS = $(MODULE_LDFLAGS) $(PYTHON2_LDADD)
+modules_python_libmod_python_la_DEPENDENCIES = $(MODULE_DEPS_LIBS)
+
```

Comment:
With this change I would try to make the Python version explicit in our python module naming. Since we have module auto-loading, the changing of the module name shouldn't matter from a user's perspective.

e.g. we should never generate an unversioned python module, it should be libmod-python2 and libmod-python3. At the very least this should resolve potential file naming conflicts.

I would also remove ENABLE_PYTHONv2_AND_v3 conditional and only use ENABLE_PYTHON2 and ENABLE_PYTHON3, and maybe couple the same with configure command line changes (e.g. separate --enable-python2 and --enable-python3 options), with both autodetected by default.

Then the only point for potential conflict would be our use of the "python" keyword, e.g. which version should be picked when the user uses the python keyword, which has been our only keyword in the past. I have the following ideas:

1) use a configure option and specify versions explicitly for this case  and use the version we are compiled with on the prominent platforms (for instance CzanikP's RHEL6 packages are compiled with Python3), 
2) pick the one that the distro uses when launching the unversioned python (can be detected at configure time)
3) the same but also honour distro configuration at runtime (IIRC Fedora allows the default to be changed)

Maybe some of these can be combined without getting insane and getting a lot of complexity and least amount of breakage down the line.